Add v18 dates

thesmiler opened the pull request:
## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [X] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: N/A


## What is the new behavior?


## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

thesmiler committed:
docs: add upcoming release dates

Adds the tentatively planned release weeks for 17.1-18.0 - please note
that these might change (and 17.3 might or might not be released).

thesmiler committed:
docs: remove extra spare line

removed accidentially included line

thesmiler committed:
docs: fixed the year

Set the year of 17.1-18.0 releases to 2024

JeanMeche commented:
Can you do the same for adev @thesmiler in `adev/src/content/reference/releases.md` thx !

JeanMeche commented on aio/content/guide/releases.md:
```diff
@@ -112,9 +112,10 @@ Approximate dates are offered as general guidance and are subject to change.
-| v16.1   | Week of 2023-06-12 |
-| v16.2   | Week of 2023-08-07 |
-| v17.0   | Week of 2023-11-06 |
+| v17.1   | Week of 2024-01-08 |
```
It's likely a week later because of the holiday break.

thesmiler replied:
We haven't quite decided this yet. But will change to make it more conservative.

thesmiler committed:
docs: Add v18 dates

Add the v18 dates to a.dev as well and shift v17.1 to a more
conservative date (exact week is tbc)

thesmiler commented:
> Can you do the same for adev @thesmiler in `adev/src/content/reference/releases.md` thx !
done.

josephperrott reviewed (approved):
LGTM

Reviewed-for: global-approvers

atscott commented:
This PR was merged into the repository by commit 00cbdc99fd8e037e55d8f9912088258d233331dc.
